app/routes/preprints/provider.js

```diff
@@ -32,6 +32,10 @@ export default Ember.Route.extend({
     },
 
     afterModel(model, transition) {
-        if (!model.get('reviewsWorkflow') && transition.targetName !== 'preprints.provider.setup') return this.replaceWith('preprints.provider.setup', model);
+        if (!model.get('permissions').contains('view_submissions')) {
+            this.replaceWith('forbidden');
+        } else if (!model.get('reviewsWorkflow') && transition.targetName !== 'preprints.provider.setup') {
+            return this.replaceWith('preprints.provider.setup', model);
```

Comment:
`replaceWith` returns the transition object... does returning that from `afterModel` do anything?

Reply:
Yup, `afterModel` is called with a transition object. See [Transitions docs](https://guides.emberjs.com/v2.15.0/routing/preventing-and-retrying-transitions/#toc_aborting-transitions-within-code-model-code-code-beforemodel-code-code-aftermodel-code) or [Redirection docs](https://guides.emberjs.com/v2.15.0/routing/redirection/).

Reply:
I'm wondering why there's a `return` here, but not two lines up. It [looks like](https://www.emberjs.com/api/ember/2.15/classes/Ember.Route/methods/afterModel?anchor=afterModel) values returned from `afterModel` are ignored unless they're promises.

Reply:
Ohhh, I get what you're saying now. I didn't add that second line but it does work. I guess it be useful for the transition to wait for it to finish.

Reply:
I'm pretty sure returning does nothing, and I feel like it's misleading to include. Would you mind removing it?